src/experimental/mdspan

```diff
@@ -56,9 +56,13 @@
 #pragma clang diagnostic ignored "-Wreserved-id-macro"
 #endif
 
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Werror=undef"
```

Comment:
This should be `-Wundef`.  

The error message is confusing because we also have `-Werror` that converts warning to errors for the CI to capture.